web/src/OverviewSidebarOptions.test.tsx

```diff
@@ -199,6 +203,53 @@ describe("overview sidebar options", () => {
     expect(pinned).toHaveLength(1)
     expect(pinned.at(0).props().item.name).toEqual("beep")
   })
+
+  it("applies the name filter", () => {
+    // 'B p' tests both case insensitivity and a multi-term query
+    sidebarOptionsAccessor.set({ ...defaultOptions, resourceNameFilter: "B p" })
+    const root = mount(
+      <MemoryRouter>
+        <tiltfileKeyContext.Provider value="test">
+          <SidebarPinContextProvider>
+            {TwoResourcesTwoTests()}
+          </SidebarPinContextProvider>
+        </tiltfileKeyContext.Provider>
+      </MemoryRouter>
+    )
+
+    assertSidebarItemsAndOptions(
+      root,
+      ["beep", "boop"],
+      defaultOptions.testsHidden,
+      defaultOptions.testsOnly,
+      defaultOptions.alertsOnTop,
+      "B p"
+    )
+  })
+
+  it("says no matches found", () => {
+    sidebarOptionsAccessor.set({
+      ...defaultOptions,
+      resourceNameFilter: "asdfawfwef",
```

Comment:
what a good search term

Reply:
carefully crafted to ensure it did not match any of the test data

Reply:
bespoke, artisanal test cases